frontend/packages/dev-console/src/components/topology/application-panel/__tests__/ApplicationGroupResource.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { Link } from 'react-router-dom';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import TopologyApplicationResourceList from '../TopologyApplicationList';
+import ApplicationGroupResource from '../ApplicationGroupResource';
+import { sampleDeployments } from '../../__tests__/topology-test-data';
+
+describe(ApplicationGroupResource.displayName, () => {
+  it('should component exists', () => {
+    const wrapper = shallow(
+      <ApplicationGroupResource
+        title="Deployments"
+        resourcesData={sampleDeployments.data}
+        group="a"
+      />,
+    );
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  it('should not exists when resourceData is an empty array', () => {
+    const wrapper = shallow(
+      <ApplicationGroupResource title="Deployments" resourcesData={[]} group="a" />,
+    );
+    expect(wrapper.find('div').exists()).toBe(false);
```

Comment:
Same comment as above, but inverse it:

```suggestion
    expect(wrapper.isEmptyRender()).toBe(true);
```

It should be noted you should also try to avoid markup like `div` as if we ever change it to be something else it'll mess with this test.

Reply:
I noticed this may not work because of the fragment... not sure why there is a fragment in the first place either. So you'll have to remove the fragment logic and use this above test or find a different way to test without markup being part of your test.

Reply:
fixed